services/workflows-service/src/alert/alert.service.ts

```diff
@@ -234,24 +241,72 @@ export class AlertService {
         dataTimestamp: new Date(),
         state: AlertState.triggered,
         status: AlertStatus.new,
-        executionDetails: {},
-        ...Object.assign({}, ...(data || [])),
+        executionDetails: {
+          checkpoint: {
+            hash: computeHash(executionRow),
+            transactionCount: findByKeyCaseInsensitive(executionRow, 'transactionCount'),
+          },
+          executionRow,
+        } as TExecutionDetails,
+        ...Object.assign({}, ...(subject || [])),
       },
     });
   }
 
   private async isDuplicateAlert(
-    alertDef: AlertDefinition,
-    data: Array<{ [key: string]: unknown }>,
+    alertDefinition: AlertDefinition,
+    subjectPayload: Array<Record<string, unknown>>,
+    executionRow: Record<string, unknown>,
   ): Promise<boolean> {
-    return await this.alertRepository.exists(
+    if (!alertDefinition.dedupeStrategy) {
+      return false;
+    }
+
+    const dedupeStrategy = alertDefinition.dedupeStrategy as TDedupeStrategy;
+
+    if (dedupeStrategy.mute) {
+      return true;
+    }
+
+    // TODO: Fix types for jsonb fields
+    const { cooldownTimeframeInMinutes } = dedupeStrategy || DEFAULT_DEDUPE_STRATEGIES;
+
+    const existingAlert = await this.alertRepository.findFirst(
       {
         where: {
-          AND: [{ alertDefinitionId: alertDef.id }, ...data],
+          AND: [{ alertDefinitionId: alertDefinition.id }, ...subjectPayload],
         },
       },
-      [alertDef.projectId],
+      [alertDefinition.projectId],
     );
+
+    if (!existingAlert) {
+      return false;
+    }
+
+    // We already have an alert for the same definition and subject, check if it's within the cooldown period
+
+    // Convert minutes to milliseconds
```

Comment:
Unnecessary comment. The variable names are pretty clear and it's easy to understand.